src/binary/ibc-setup/commands/channel.ts

```diff
@@ -0,0 +1,154 @@
+import path from 'path';
+
+import { Order } from '../../../codec/ibc/core/channel/v1/channel';
+import { Link } from '../../../lib/link';
+import { registryFile } from '../../constants';
+import { Logger } from '../../create-logger';
+import { loadAndValidateApp } from '../../utils/load-and-validate-app';
+import { loadAndValidateRegistry } from '../../utils/load-and-validate-registry';
+import { resolveOption } from '../../utils/options/resolve-option';
+import { resolveHomeOption } from '../../utils/options/shared/resolve-home-option';
+import { resolveKeyFileOption } from '../../utils/options/shared/resolve-key-file-option';
+import { resolveMnemonicOption } from '../../utils/options/shared/resolve-mnemonic-option';
+import { signingClient } from '../../utils/signing-client';
+
+export type Flags = {
+  readonly interactive: boolean;
+  readonly ordered: boolean;
+  readonly mnemonic?: string;
+  readonly keyFile?: string;
+  readonly home?: string;
+  readonly src?: string;
+  readonly dest?: string;
+  readonly srcConnection?: string;
+  readonly destConnection?: string;
+  readonly srcPort?: string;
+  readonly destPort?: string;
+  readonly version?: string;
+};
+
+export type Options = {
+  readonly home: string;
+  readonly mnemonic: string;
+  readonly src: string;
+  readonly dest: string;
+  readonly srcConnection: string;
+  readonly destConnection: string;
+  readonly srcPort: string;
+  readonly destPort: string;
+  readonly version: string;
+  readonly ordered: boolean;
+};
+
+export async function channel(flags: Flags, logger: Logger) {
+  const home = resolveHomeOption({ homeFlag: flags.home });
+  const app = loadAndValidateApp(home);
+
+  const keyFile = resolveKeyFileOption({ keyFileFlag: flags.keyFile, app });
+  const mnemonic = await resolveMnemonicOption({
+    interactiveFlag: flags.interactive,
+    mnemonicFlag: flags.mnemonic,
+    keyFile: keyFile,
+    app,
```

Comment:
Let's be consistent with use of shorthand here (ideally via a lint setting).

Reply:
Alright! I added an eslint rule and fixed all the shorthand-possible occurrences.

0545096